packages/tinybased/src/lib/queries/Query.spec.ts

```diff
@@ -222,4 +222,68 @@ describe('QueryBuilder', () => {
 
     expect(rowIds).toEqual(['note3']);
   });
+
+  it('supports custom logic for where clause', async () => {
+    const db = await sb.build();
+
+    const qbForCustomWhereOnStartTable = db
+      .query('notes')
+      .select('id')
+      .select('isDraft')
+      .whereUsing((getCell) => getCell('isDraft') === true)
+      .build();
+
+    expect(qbForCustomWhereOnStartTable.getResultTable()).toEqual({
+      note1: {
+        id: 'note1',
+        isDraft: true,
+      },
+    });
+
+    const qbForCustomWhereOnJoinedTable = db
+      .query('notes')
+      .select('id')
+      .select('isDraft')
+      .join('userNotes')
+      .whereUsing((getCellFrom) => getCellFrom('users', 'name') === 'Jesse')
+      .build();
+
+    expect(qbForCustomWhereOnJoinedTable.getResultTable()).toEqual({
+      note1: {
+        id: 'note1',
+        isDraft: true,
+      },
+      note2: {
+        id: 'note2',
+        isDraft: false,
+      },
+    });
+  });
+
+  it('generates unique query ids when using whereUsing', async () => {
+    const db = await sb.build();
+    const qb1 = db
+      .query('notes')
+      .select('id')
+      .whereUsing((getCell) => (getCell('isDraft') as boolean) === true)
```

Comment:
How come this needs to be cast with `as boolean` here. Can't we infer it from the name of the cell?

Reply:
This is just for a test that generating a query id ignores typescript stuff,

Reply:
but I'm dum why ts exist in runtime, I'll remove it